### PR TITLE
board/arm/nucleo_l073rz : add entropy RNG support

### PR DESCRIPTION
**Pull request by TDhaouST**
stm32l073 add entropy RNG testcase running on nucleo target

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
Please fix commit message:
"stm32l073 add entropy RNG testcase running on nucleo target" is not exactly what you're doing here.

**Review by FRASTM — COMMENTED** [review]
also add the update of
-  ./boards/arm/nucleo_l073rz/doc 
- ./boards/arm/nucleo_l073rz/nucleo_l073rz.yaml

**Review by erwango — DISMISSED** [review]

File: boards/arm/nucleo_l073rz/nucleo_l073rz.dts [hunk 1]
```diff
@@ -126,3 +126,7 @@
 		pinctrl-names = "default";
 	};
 };
+
+&rng {
```
Comment (gmarull) [inline boards/arm/nucleo_l073rz/nucleo_l073rz.dts:130]: Maybe add `zephyr,entropy` choice as well.

Comment (TDhaouST) [reply boards/arm/nucleo_l073rz/nucleo_l073rz.dts:131]: yes, it's done

**Review by gmarull — DISMISSED** [review]

File: boards/arm/nucleo_l073rz/doc/index.rst [hunk 2]
```diff
@@ -59,6 +59,8 @@ Nucleo L073RZ provides the following hardware components:
   - Mass storage
   - Debug port
 
+- True random number generator
```
Comment (ABOSTM) [inline boards/arm/nucleo_l073rz/doc/index.rst:62]: This is the list of Hardware component provided by the board.
But RNG is not provided by the board  (rather by the MCU)
so there is no reason to add RNG in this list

**Review by ABOSTM — CHANGES_REQUESTED** [review]

**Review by ABOSTM — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**